### PR TITLE
feat: support by-dx legends, non-sum aggtypes, sortable columns, and ou hierarchies

### PR DESCRIPTION
**Pull request by amcgee**
More testing is needed for aggtypes like `count`, `variance`, `avg by org unit`, `std dev`, etc. (I *think* they should work) but I've tried to match the previous engine's behavior from `d2-analysis` for now, so it should at least be consistent.

`PivotTableEngine.js` is getting pretty large (800 lines) and unwieldy, will take a pass at re-factoring it soon.

### CODE REVIEW

File: src/modules/pivotTable/PivotTableEngine.js [hunk 1]
```diff
@@ -103,6 +119,23 @@ const buildDimensionLookup = (visualization, metadata, headers) => {
         return out
     }, {})
 
+    const primaryDimensions = {
+        data: getPrimaryDimension(metadata, 'dx'),
+        period: getPrimaryDimension(metadata, 'pe'),
+        orgUnit: getPrimaryDimension(metadata, 'ou'),
```
Comment (jenniferarnesen) [inline src/modules/pivotTable/PivotTableEngine.js]: Is it worth using the constants for 'dx', 'pe', 'ou' that are defined here: modules/predefinedDimensions.js?

**Comment by jenniferarnesen** [comment]
This PR is quite big and thus difficult to say much about. I guess at this point it is hands on testing and exposure that is needed, so I would say - get it merged ASAP to get people trying it out.

When there is time to go back and revisit after the release, my suggestions would be refactoring the PivotTableEngine (as Austin has already mentioned), add unit tests at least for the modules, and possibly some code documentation of PivotTableEngine.

File: src/modules/pivotTable/PivotTableEngine.js (hunk 1 above)
Comment (amcgee) [reply src/modules/pivotTable/PivotTableEngine.js]: Sure, will do

Comment (amcgee) [reply src/modules/pivotTable/PivotTableEngine.js]: Done

**Comment by amcgee** [comment]
Thanks @jenniferarnesen !  I completely agree we should get it in for testing (and we can iterate if there are any remaining bugs to track down).  If @janhenrikoverland is OK with that, can we get it in tonight so we can look at #326  #327 and #328 which depend on this?

**Review by jenniferarnesen — APPROVED** [review]

**Comment by janhenrikoverland** [comment]
+1

**Pull request merged**